scripts/extract_download_logs.py

```diff
@@ -3,45 +3,68 @@
 For script options, run the script with '--help' argument.
 """
 
+import argparse
 import csv
 import datetime
+import io
 import json
+import os
 import time
-from optparse import OptionParser
+from io import StringIO
 from typing import List
 
 from boto3 import client
+from dateutil.relativedelta import relativedelta
+from notifications_python_client import prepare_upload
+from notifications_python_client.notifications import NotificationsAPIClient
 
-parser = OptionParser(
-    usage="Output a report of downloads (requires AWS authentication)"
+parser = argparse.ArgumentParser(
+    description="Output a report of downloads (requires AWS authentication)",
 )
-parser.add_option(
+
+time_option_group = parser.add_mutually_exclusive_group()
+time_option_group.add_argument(
+    "-d",
+    "--days",
+    dest="days",
+    type=int,
+    help="Specify the number of days",
```

Comment:
This help text could be improved to make it more explicit e.g., "Specify the number of days to include in the report, counting backwards from today. This option generates a report covering the specified period."

Same with the months arg

Reply:
Have updated as suggested in: 90ba4b7